src/mobile/apps/artwork/components/image/view.coffee

```diff
@@ -37,7 +37,7 @@ module.exports = class ArtworkImageView extends Backbone.View
               'data-state': 'saved'
               'data-action': 'remove'
     else
-      @$('.artwork-header-module__favorites-container a').attr("href", "/sign_up?action=artwork-save&redirect-to=#{window.location}")
+      @$('.artwork-header-module__favorites-container a').attr("href", "/sign_up?action=artwork-save&entity-id=#{@artwork.id}&redirect-to=#{window.location}")
```

Comment:
I think we might want to change the action property value to just be `save` versus `artwork-save`.  This was previously used to determine which template to render and now we are using it to handle the `afterSignUpAction` which [specifically checks](https://github.com/artsy/force/blob/master/src/mobile/components/layout/bootstrap.coffee#L93-L97) for `save` or `follow`.

Reply:
Yup, I just updated all this and fixed it in my last commit